lib/rester/client/adapters/stub_adapter.rb

```diff
@@ -82,6 +84,27 @@ def _process_request(path, verb, params)
       end
 
       ##
+      # Generate the diff string in the case when the request params of the
+      # service don't match the params specified in the stub file.
+      def _param_diff(params, spec_params)
+        # Compile a list of mismatched params values
+        diff = spec_params.map { |k,v|
+          param_value = params.delete(k)
+          unless v == param_value
+            "#{k.inspect} should equal #{v.inspect} but got #{param_value.inspect}"
+          end
+        }.compact.join(', ')
+
+        unless params.empty?
+          # Add any param keys which aren't specified in the spec
+          diff << '. ' unless diff.empty?
+          diff << "Unexpected key(s): #{params.keys}"
+        end
```

Comment:
Then you'd get messages like:
 * `... params don't match stub: received unexpected key(s): "key1"`
 * `... params don't match stub: "key1" should equal "1234" but got nil, and received unexpected key(s): "key2"`